floating_recipient_bar improvements and fix date <span> id duplication

dhcrzf opened the pull request:
Force display of the top-most recipient_bar's recipient_row_date
when the floating_recipient_bar is just about to overlap and
becomes hidden while user is scrolling.

Fixes #4844

WIP while also working to fix #4997 and #5128.

smarx commented:
*Automated message from Dropbox CLA bot*

@dhcrzf, it looks like you've already signed the Dropbox CLA. Thanks!

dhcrzf commented:
Added commits to fix #4997 and #5128.

With these changes, the timerender date spans will be identified by class="timerender0" rather than id="timerender0". The cloned span (floating bar) will share the same class as the source span, leading timerender.update_timestamps() to update both date spans at the same time.

I also added some minor refactoring. The array of timestamps to be updated by timerender.update_timestamps() now contains objects instead of arrays. We made similar changes to timerender.render_now() in PR #4833.

timabbott commented on static/js/timerender.js:
```diff
@@ -113,50 +113,54 @@ function render_date_span(elem, rendered_time, rendered_time_above) {
-                maybe_add_update_list_entry(rendered_time.needs_update, id, time, time_above);
+            if (elements !== null) {
+                _.each(elements, function (element) {
+                  blueslip.log(element);
```
I think this blueslip.log should be removed in this commit, not the next one.

timabbott commented:
Merged, after fixing the one style issue I mentioned above.

Thanks @dhcrzf!  If you're interested in doing more with timerender, the other thing that would be awesome to do is getting node test coverage to 100% (we have a push being coordinated on the "#test suites" stream in chat.zulip.org).
